src/Promitor.Core.Scraping/AzureMonitorScraper.cs

```diff
@@ -45,27 +46,27 @@ protected override async Task<ScrapeResult> ScrapeResourceAsync(string subscript
             var metricLimit = DetermineMetricLimit(scrapeDefinition);
 
             // Determine the metric dimension to use, if any
-            var dimensionName = DetermineMetricDimension(metricName, resourceDefinition, scrapeDefinition.AzureMetricConfiguration?.Dimension);
+            var dimensionNames = DetermineMetricDimensions(metricName, resourceDefinition, scrapeDefinition.AzureMetricConfiguration?.Dimensions);
 
-            List<MeasuredMetric> measuredMetrics = new List<MeasuredMetric>();
+            var measuredMetrics = new List<MeasuredMetric>();
             try
             {
                 // Query Azure Monitor for metrics
-                measuredMetrics = await AzureMonitorClient.QueryMetricAsync(metricName, dimensionName, aggregationType, aggregationInterval, resourceUri, metricFilter, metricLimit);
+                measuredMetrics = await AzureMonitorClient.QueryMetricAsync(metricName, dimensionNames, aggregationType, aggregationInterval, resourceUri, metricFilter, metricLimit);
             }
             catch (MetricInformationNotFoundException metricsNotFoundException)
             {
-                Logger.LogWarning("No metric information found for metric {MetricName} with dimension {MetricDimension}. Details: {Details}", metricsNotFoundException.Name, metricsNotFoundException.Dimension, metricsNotFoundException.Details);
+                Logger.LogWarning("No metric information found for metric {MetricName} with dimensions {MetricDimensions}. Details: {Details}", metricsNotFoundException.Name, metricsNotFoundException.Dimensions, metricsNotFoundException.Details);
                 
-                var measuredMetric = string.IsNullOrWhiteSpace(dimensionName) ? MeasuredMetric.CreateWithoutDimension(null) : MeasuredMetric.CreateForDimension(null, dimensionName, "unknown");
+                var measuredMetric = dimensionNames.Any() ? MeasuredMetric.CreateForDimension(null, dimensionNames, Enumerable.Repeat("unknown",dimensionNames.Count).ToList()) : MeasuredMetric.CreateWithoutDimension(null);
```

Comment:
We can also handle this in the constructor if need be
```suggestion
                var measuredMetric = dimensionNames.Any() ? MeasuredMetric.CreateForDimension(null, dimensionNames, "unknown") : MeasuredMetric.CreateWithoutDimension(null);
```

Reply:
I like that idea as well, especially with the new way of having dimension names and values combined in one list.
Let me know how you feel about me adding an overload only requiring the `dimensionNames` for this behaviour.

Reply:
Sounds good!